src/baseTrie.js

```diff
@@ -49,6 +49,38 @@ module.exports = class Trie {
     this.root = root
   }
 
+  static fromProof(proofNodes, cb, proofTrie){
```

Comment:
what you are making here could be called a _sparse trie_ . optional proofTrie argument so you can build onto an existing sparse trie is you want.